src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/InputAllowNonOverlappingLocalVars.java

```diff
@@ -0,0 +1,49 @@
+/*
+Input test file for RequireThisCheck.
+Created: 2017
+ */
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+class InputAllowNonOverlappingLocalVars {
+
+    private String s1 = "foo1";
+    String s2 = "foo2";
+
+    InputAllowNonOverlappingLocalVars() {
+        s1 = "bar1"; // Violation. Requires "this".
+        String s2;
+        s2 = "bar2"; // No violation. Non-overlapping overshadow by local var allowed.
```

Comment:
I was using overshadow to denote for example when a local var. in a method would have the same name as an instance variable (overshadow it). I was using overlapping only when a local var and an instance var with the same name are there in the same expression (overlapped with each other). But yes you guys are correct it adds confusion to the mix ! Should I remove the word overshadow from everywhere ? or should I somewhere mention the intended difference between overshadowing and overlapping ? 
(I feel it would be alright to remove)
Thanks